extensions/elation/actions/updateNonVisitNote.ts

```diff
@@ -0,0 +1,172 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { ZodError } from 'zod'
+import { FieldType, type Action, type Field } from '../../../lib/types'
+import { Category } from '../../../lib/types/marketplace'
+import { type settings } from '../settings'
+import { makeAPIClient } from '../client'
+import { fromZodError } from 'zod-validation-error'
+import { AxiosError } from 'axios'
+import { nonVisitNoteSchema } from '../validation/nonVisitNote.zod'
+import { NumericIdSchema } from '../../../lib/shared/validation'
+import { isNil } from 'lodash'
+
+/**
+ * Although fields are not required, editing of e.g. bullet text requires providing a bullet id and an author
+ */
+const fields = {
+  nonVisitNoteId: {
+    id: 'nonVisitNoteId',
+    label: 'Non-Visit Note ID',
+    description: 'ID of a note',
+    type: FieldType.NUMERIC,
+    required: true,
+  },
+  nonVisitNoteBulletId: {
+    id: 'nonVisitNoteBulletId',
+    label: 'Non-Visit Note Bullet ID',
+    description: 'ID of a bullet in a note',
+    type: FieldType.NUMERIC,
+    required: false,
+  },
+  text: {
+    id: 'text',
+    label: 'Text',
+    description: 'Text of a note',
+    type: FieldType.STRING,
+    required: false,
+  },
+  author: {
+    id: 'author',
+    label: 'Author',
+    description: 'Author of a note. Should be ID of a User.',
+    type: FieldType.NUMERIC,
+    required: false,
+  },
+  patient: {
+    id: 'patient',
+    label: 'Patient',
+    description: 'ID of a Patient',
+    type: FieldType.NUMERIC,
+    required: false,
+  },
+  category: {
+    id: 'category',
+    label: 'Category',
+    description:
+      'Category of a note. Defaults to "Problem". One from the list: "Problem", "Past", "Family", "Social", "Instr", "PE", "ROS", "Med", "Data", "Assessment", "Test", "Tx", "Narrative", "Followup", "Reason", "Plan", "Objective", "Hpi", "Allergies", "Habits", "Assessplan", "Consultant", "Attending", "Dateprocedure", "Surgical", "Orders", "Referenced", "Procedure".',
+    type: FieldType.STRING,
+    required: false,
+  },
+  practice: {
+    id: 'practice',
+    label: 'Practice',
+    description: 'ID of a Practice',
+    type: FieldType.NUMERIC,
+    required: false,
+  },
+  documentDate: {
+    id: 'documentDate',
+    label: 'Document Date',
+    description: 'Date in ISO 8601 format',
+    type: FieldType.DATE,
+    required: false,
+  },
+  chartDate: {
+    id: 'chartDate',
+    label: 'Chart Date',
+    description: 'Date in ISO 8601 format',
+    type: FieldType.DATE,
+    required: false,
+  },
+  tags: {
+    id: 'tags',
+    label: 'Tags',
+    description: 'Comma-separated list of tags IDs',
+    type: FieldType.STRING,
+    required: false,
+  },
+} satisfies Record<string, Field>
+
+export const updateNonVisitNote: Action<typeof fields, typeof settings> = {
+  key: 'updateNonVisitNote',
+  category: Category.EHR_INTEGRATIONS,
+  title: 'Update Non-Visit Note',
+  description: "Update a Non-Visit Note using Elation's patient API.",
```

Comment:
your comments about "bullet ID is required" is probably important for the care flow builder to know. maybe you can include that important information in the description of the action, or in the description of the field?

Reply:
I did both. In a field description and action description